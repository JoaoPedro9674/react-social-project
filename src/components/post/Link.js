import React, { PropTypes } from 'react';

const Link = (props) => {
  const { link } = props;
  return (
    link ?
      <div className="link">
        <a className="title" href={link.url}>
          <h4>{link.title}</h4>
        </a>
        <div className="url">{link.url}</div>
        <p className="description">{link.description}</p>
      </div>
    :
    null
  );
};

Link.propTypes = {
  link: PropTypes.shape({
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  }),
};

export { Link };
