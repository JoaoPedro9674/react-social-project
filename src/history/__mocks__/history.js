import { stub } from 'sinon';

export const history = stub.yield();
export const navigate = to => stub(to);
