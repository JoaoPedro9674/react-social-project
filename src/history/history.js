import createHistory from 'history/createBrowserHistory';

const history = typeof window !== 'undefined'
    ? createHistory()
    : { push: () => {} };
const navigate = to => history.push(to);

export { history, navigate };
