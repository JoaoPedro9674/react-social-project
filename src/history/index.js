export * from './history';
export * from './navigation';
