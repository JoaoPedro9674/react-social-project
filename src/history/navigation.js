import { history } from './';

export const navigate = to => history.push(to);
