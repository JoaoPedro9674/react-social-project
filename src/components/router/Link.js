import { PropTypes, Children, Component, cloneElement } from 'react';
import { navigate } from '../../history';

class Link extends Component {
  static propTypes = {
    to: PropTypes.string,
    children: PropTypes.node,
  }

  render() {
    const { to, children } = this.props;
    return cloneElement(Children.only(children), {
      onClick: () => navigate(to),
    });
  }
}

export { Link };
