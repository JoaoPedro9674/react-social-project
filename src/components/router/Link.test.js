import React from 'react';
import { spy } from 'sinon';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { history } from '../../history';
import { Link } from './Link';
jest.mock('../../history');

describe('<Link/>', () => {
    const ChildStub = () => <div>inner child</div>;
    describe('render methods', () => {
        it('should return an element with an onClick method attached', () => {
            const wrapper = shallow(
                <Link to="/">
                    <ChildStub/>
                </Link>
            );
            expect(wrapper.find(ChildStub).length).toBe(1);
            expect(wrapper.props().onClick).toBeDefined();
        });
        it('should render correctly', () => {
          const component = renderer.create(
            <Link to="/">
                <ChildStub/>
            </Link>
          );
          const tree = component.toJSON();
          expect(tree).toMatchSnapshot();
        });
    });
});
