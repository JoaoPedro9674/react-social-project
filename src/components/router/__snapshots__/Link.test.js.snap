exports[`<Link/> render methods should render correctly 1`] = `
<div>
  inner child
</div>
`;
